spec/integration_spec.rb

```diff
@@ -359,167 +355,6 @@
   end
 end
 
-describe 'Book' do
-  before do
-    Book.clear_index!(true)
-    Book.index(safe_index_uid('BookAuthor')).delete_all_documents
-    Book.index(safe_index_uid('Book')).delete_all_documents
-  end
-
-  it 'returns array of tasks on #ms_index!' do
-    moby_dick = Book.create! name: 'Moby Dick', author: 'Herman Melville', premium: false, released: true
-
-    tasks = moby_dick.ms_index!
-
-    expect(tasks).to contain_exactly(
-      a_hash_including('uid'),
-      a_hash_including('taskUid'),
-      a_hash_including('taskUid')
-    )
-  end
-
-  it 'indexes the book in 2 indexes of 3' do
-    steve_jobs = Book.create! name: 'Steve Jobs', author: 'Walter Isaacson', premium: true, released: true
-    results = Book.search('steve')
-    expect(results.size).to eq(1)
-    expect(results).to include(steve_jobs)
-
-    index_author = Book.index(safe_index_uid('BookAuthor'))
-    expect(index_author).not_to be_nil
-    results = index_author.search('steve')
-    expect(results['hits'].length).to eq(0)
-    results = index_author.search('walter')
-    expect(results['hits'].length).to eq(1)
-
-    # premium -> not part of the public index
-    index_book = Book.index(safe_index_uid('Book'))
-    expect(index_book).not_to be_nil
-    results = index_book.search('steve')
-    expect(results['hits'].length).to eq(0)
-  end
-
-  it 'sanitizes attributes' do
-    _hack = Book.create! name: '"><img src=x onerror=alert(1)> hack0r',
-                         author: '<script type="text/javascript">alert(1)</script>', premium: true, released: true
-    b = Book.raw_search('hack', { attributes_to_highlight: ['*'] })
-    expect(b['hits'].length).to eq(1)
-    begin
-      expect(b['hits'][0]['name']).to eq('"> hack0r').and_raise(StandardError)
-      expect(b['hits'][0]['author']).to eq('alert(1)')
-      expect(b['hits'][0]['_formatted']['name']).to eq('"> <em>hack</em>0r')
-    rescue StandardError
-      # rails 4.2's sanitizer
-      begin
-        expect(b['hits'][0]['name']).to eq('&quot;&gt; hack0r').and_raise(StandardError)
-        expect(b['hits'][0]['author']).to eq('')
-        expect(b['hits'][0]['_formatted']['name']).to eq('&quot;&gt; <em>hack</em>0r')
-      rescue StandardError
-        # jruby
-        expect(b['hits'][0]['name']).to eq('"&gt; hack0r')
-        expect(b['hits'][0]['author']).to eq('')
-        expect(b['hits'][0]['_formatted']['name']).to eq('"&gt; <em>hack</em>0r')
-      end
-    end
-  end
-
-  it 'handles removal in an extra index' do
```

Comment:
Moved to `record_is_updated_spec.rb` since it tests a callback on an event (record being updated)